src/graph/strongly-connected-components.md

```diff
@@ -4,151 +4,148 @@ tags:
+            dfs2(v);
+            int root = component.front();
+            for (auto u : component) roots[u] = root;
+            root_nodes.push_back(root);
             component.clear();
         }
+    
+    
+    // add edges to condensation graph
+    for (int v = 0; v < n; v++)
+        for (auto u : adj[v])
+            if (roots[v] != roots[u])
+                adj_scc[roots[v]].push_back(roots[u]);
+
+    // finished finding SCCs and constructing condensation graph.
+    .......
 }
 ```
 
-Here, $g$ is graph, $gr$ is transposed graph. Function $dfs1$ implements depth first search on graph $G$, function $dfs2$ - on transposed graph $G^T$. Function $dfs1$ fills the list $order$ with vertices in increasing order of their exit times (actually, it is making a topological sort). Function $dfs2$ stores all reached vertices in list $component$, that is going to store next strongly connected component after each run.
-
-### Condensation Graph Implementation
-
-```cpp
-// continuing from previous code
-
-vector<int> roots(n, 0);
-vector<int> root_nodes;
-vector<vector<int>> adj_scc(n);
-
-for (auto v : order)
-    if (!used[v]) {
-        dfs2(v);
-
-        int root = component.front();
-        for (auto u : component) roots[u] = root;
-        root_nodes.push_back(root);
-
-        component.clear();
-    }
-
-
-for (int v = 0; v < n; v++)
-    for (auto u : adj[v]) {
-        int root_v = roots[v],
-            root_u = roots[u];
-
-        if (root_u != root_v)
-            adj_scc[root_v].push_back(root_u);
-    }
-```
+Here, the function `dfs1` implements depth first search on $G$, and the function `dfs2` implements depth first search on $G^T$. The function `dfs1` fills the vector `order` with vertices in increasing order of their exit times. The function `dfs2` adds all reached vertices to the vector `component`, which, after each run, will contain the just-found strongly connected component.
 
-Here, we have selected the root of each component as the first node in its list. This node will represent its entire SCC in the condensation graph. `roots[v]` indicates the root node for the SCC to which node `v` belongs. `root_nodes` is the list of all root nodes (one per component) in the condensation graph. 
+When building the condensation graph, we select the *root* of each component as the first node in its list (this is an arbitrary choice). This node will represent its entire SCC in the condensation graph. For each vertex `v`, the value `roots[v]` indicates the root node of the SCC which `v` belongs to. `root_nodes` is the list of all root nodes (one per component) in the condensation graph. 
 
-`adj_scc` is the adjacency list of the `root_nodes`. We can now traverse on `adj_scc` as our condensation graph, using only those nodes which belong to `root_nodes`.
+Our condensation graph is now given by the vertices `root_nodes`, and the adjacency list is given by `adj_scc`, using only those vertices which belong to `root_nodes`.
```

Comment:
Maybe worth noting that adjacency list may contain multiple edges between components.

Reply:
Indeed, thanks, I was a bit too hasty when removing that sentence. Resolved in [88eabab](https://github.com/cp-algorithms/cp-algorithms/pull/1307/commits/88eababe2aa10a31575deb998a67b2079df2f62d).